std/array.d

```diff
@@ -3891,3 +3891,231 @@ unittest
++/
+pragma(inline, true) T[n] asStatic(T, size_t n)(auto ref T[n] arr) @nogc
+{
+    return arr;
+}
+
+/// ditto
+U[n] asStaticCast(U, T, size_t n)(auto ref T[n] arr) @nogc
+{
+    U[n] ret = void;
+    static foreach (i; 0 .. n)
+    {
+        emplaceRef!U(ret[i], cast(U) arr[i]);
+    }
+    return ret;
+}
+
+///
+@safe unittest
+{
+    int val = 3;
+    static immutable gold = [1, 2, 3];
+    [1, 2, val].asStatic.assertSame!int([1, 2, 3]);
+
+    @nogc void checkNogc()
+    {
+        [1, 2, val].asStatic.assertSame!int(gold);
+    }
+
+    [1, 2, val].asStaticCast!double.assertSame!double(gold);
+    [1, 2, 3].asStaticCast!int.assertSame!int(gold);
+
+    [1, 2, 3].asStaticCast!(const(int)).assertSame!(const(int))(gold);
+    [1, 2, 3].asStaticCast!(const(double)).assertSame!(const(double))(gold);
+    {
+        const(int)[3] a2 = [1, 2, 3].asStatic;
+    }
+
+    [1, 129].asStaticCast!byte.assertSame!byte([1, -127]);
+
+    version (Bug)
+    {
+        // https://issues.dlang.org/show_bug.cgi?id=16779
+        auto a2 = [1, 2, 3].asStatic!byte;
+        auto a3 = [1, 2, 3].asStatic!ubyte;
+
+        // NOTE: correctly issues a deprecation
+        int[] a2 = [1, 2].asStatic;
+    }
+}
+
+/// ditto but with an range
+auto asStatic(size_t n, T)(T a) @nogc
+{
+    // TODO: ElementType vs ForeachType
+    alias U = typeof(a[0]);
+    U[n] ret = void;
+    size_t i;
+    foreach (ref ai; a)
+    {
+        emplaceRef!U(ret[i++], ai);
+    }
+    assert(i == n);
+    return ret;
+}
+
+/// ditto
+auto asStaticCast(Un : U[n], U, size_t n, T)(T a) @nogc
+{
+    U[n] ret = void;
+    size_t i;
+    foreach (ref ai; a)
+    {
+        emplaceRef!U(ret[i++], cast(U) ai);
+    }
+    assert(i == n);
+    return ret;
+}
+
+///
+@safe unittest
+{
+    import std.range;
+
+    2.iota.asStatic!2.assertSame!int([0, 1]);
+    2.iota.asStaticCast!(double[2]).assertSame!double([0, 1]);
+    2.iota.asStaticCast!(byte[2]).assertSame!byte([0, 1]);
+}
+
+///
+@system unittest
+{
+    import std.range;
+
+    assert(isThrown!Error(2.iota.asStatic!1));
+    assert(isThrown!Error(2.iota.asStatic!3));
+}
+
+/// ditto but with an alias
+auto asStatic(alias arr)() @nogc
+{
+    enum n = arr.length;
+    alias U = typeof(arr[0]);
+    U[n] ret = void;
+    static foreach (i; 0 .. n)
+    {
+        emplaceRef!U(ret[i], arr[i]);
+    }
+    return ret;
+}
+
+/// ditto
+auto asStaticCast(U, alias arr)() @nogc
+{
+    enum n = arr.length;
+    U[n] ret = void;
+    static foreach (i; 0 .. n)
+    {
+        emplaceRef!U(ret[i], cast(U) arr[i]);
+    }
+    return ret;
+}
+
+///
+@safe unittest
+{
+    import std.range;
+
+    enum a = asStatic!(2.iota);
+    asStatic!(2.iota).assertSame!int([0, 1]);
+    asStaticCast!(double, 2.iota).assertSame!double([0, 1]);
+    asStaticCast!(byte, 2.iota).assertSame!byte([0, 1]);
+}
+
+// Note: can't use `version (unittest)` because of https://issues.dlang.org/show_bug.cgi?id=18488
+void assertSame(T, T1, T2)(T1 a, T2 b) @nogc
```

Comment:
This is a new public symbol. The tester failed for a _good_ reason. We don't want to use undocumented functionality in public examples.

Reply:
removed its used from documented unittests